include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -2928,9 +2933,42 @@ namespace internal
       Number *temp         = eval.get_scratch_data().begin();
       Number *scratch_data = temp + 3 * n_components * dofs_per_face;
 
-      FEFaceNormalEvaluationImpl<dim, fe_degree, Number>::template interpolate<
-        true,
-        false>(n_components, evaluation_flag, data, values_dofs, temp, face_no);
+      if (eval.get_dof_access_index() ==
+            MatrixFreeFunctions::DoFInfo::dof_access_cell &&
+          eval.get_is_interior_face() == false)
+        {
+          for (unsigned int v = 0; v < Number::size(); ++v)
+            {
```

Comment:
I guess it would be better to loop over all `2*d` faces here, instead of looping over each lanes!?

Reply:
I guess it depends how often we have a particular face number and change in orientation. I would believe that the current choice might be more efficient for moderate amounts of irregularity, as the other alternative needs to mask out some of the lanes (I think of the results in the hanging nodes algorithms).

Reply:
Looking at this again, I believe I did not state my point correctly: It is likely that it would help to detect the case where all faces have the same number, and then go to the `else` branch below, and only run the per-lane code in the other case. Alternatively, we might want to detect how many of the faces actually pop up, and then only run the code for those cases. That would be strictly less work than the loop over all lanes, as we can't have more different faces than lanes. I guess this was what you had in mind?